Source: Thomas Anderson <tanderson@exherbo.org>
Reason: Support installing to exec-prefix, useful for multiarch
Upstream: PR #119, merged to master

diff -aur exaile-3.4.5-orig/Makefile exaile-3.4.5/Makefile
--- exaile-3.4.5-orig/Makefile	2015-04-27 10:11:07.000000000 -0700
+++ exaile-3.4.5/Makefile	2015-06-27 14:34:15.211228047 -0700
@@ -1,11 +1,12 @@
 PYTHON2_CMD	?= `command -v python2`
 
 PREFIX 		?= /usr/local
+EPREFIX		?= ${PREFIX}
 LIBINSTALLDIR 	?= /lib
 XDGCONFDIR 	?= /etc/xdg
 
-EXAILEBINDIR  = ${DESTDIR}${PREFIX}/bin
-EXAILELIBDIR 	= ${DESTDIR}${PREFIX}${LIBINSTALLDIR}/exaile
+EXAILEBINDIR  = ${DESTDIR}${EPREFIX}/bin
+EXAILELIBDIR 	= ${DESTDIR}${EPREFIX}${LIBINSTALLDIR}/exaile
 EXAILESHAREDIR 	= ${DESTDIR}${PREFIX}/share/exaile
 EXAILECONFDIR 	= ${DESTDIR}${XDGCONFDIR}/exaile
 
@@ -124,10 +125,10 @@
 		${DESTDIR}${PREFIX}/share/appdata/	
 	-install -m 644 exaile.1.gz ${DESTDIR}${PREFIX}/share/man/man1/
 	install -m 644 data/config/settings.ini ${EXAILECONFDIR}
-	tools/generate-launcher "${DESTDIR}" "${PREFIX}" "${LIBINSTALLDIR}" \
+	tools/generate-launcher "${DESTDIR}" "${PREFIX}" "${EPREFIX}" "${LIBINSTALLDIR}" \
 		"${PYTHON2_CMD}" && \
-	  chmod 755 ${DESTDIR}${PREFIX}/bin/exaile
-	sed "s|\@bindir\@|${PREFIX}/bin|" data/org.exaile.Exaile.service.in > \
+	  chmod 755 ${DESTDIR}${EPREFIX}/bin/exaile
+	sed "s|\@bindir\@|${EPREFIX}/bin|" data/org.exaile.Exaile.service.in > \
 		${DESTDIR}${PREFIX}/share/dbus-1/services/org.exaile.Exaile.service
 	${MAKE} -C plugins install
 
diff -aur exaile-3.4.5-orig/tools/generate-launcher exaile-3.4.5/tools/generate-launcher
--- exaile-3.4.5-orig/tools/generate-launcher	2015-04-27 10:11:07.000000000 -0700
+++ exaile-3.4.5/tools/generate-launcher	2015-06-27 14:37:47.308600800 -0700
@@ -6,8 +6,9 @@
 
 DESTDIR=${1:-}
 PREFIX=${2:-/usr/local}
-LIBDIR=${3:-/lib}
-PYTHON2_CMD=${4:-python2}
+EPREFIX=${3:-${PREFIX}}
+LIBDIR=${4:-/lib}
+PYTHON2_CMD=${5:-python2}
 
 # realpath/readlink misbehave on OSX
 #if (command -v realpath > /dev/null); then
@@ -19,22 +20,23 @@
 #  exit 1
 #fi
 
-PREFIX=`$PYTHON2_CMD -c 'import os,sys;print os.path.realpath(sys.argv[1])' $PREFIX`
+EPREFIX=`$PYTHON2_CMD -c 'import os,sys;print os.path.realpath(sys.argv[1])' $EPREFIX`
 
-cd ${DESTDIR}${PREFIX}/bin && \
+cd ${DESTDIR}${EPREFIX}/bin && \
 cat <<EOF > exaile
 #!/bin/sh
 
 # Exaile launcher script (automatically generated)
 
 EXAILE_PREFIX=${PREFIX}
+EXAILE_EPREFIX=${EPREFIX}
 
 if test -z "\${XDG_DATA_DIRS}"; then
 	XDG_DATA_DIRS="/usr/local/share/:/usr/share/"
 fi
 XDG_DATA_DIRS="\${EXAILE_PREFIX}/share/:\${XDG_DATA_DIRS}"
 export XDG_DATA_DIRS
-exec ${PYTHON2_CMD} \${EXAILE_PREFIX}${LIBDIR}/exaile/exaile.py --datadir=\${EXAILE_PREFIX}/share/exaile/data --startgui ${DEFAULTARGS} "\$@"
+exec ${PYTHON2_CMD} \${EXAILE_EPREFIX}${LIBDIR}/exaile/exaile.py --datadir=\${EXAILE_PREFIX}/share/exaile/data --startgui ${DEFAULTARGS} "\$@"
 exit \$?
 
 EOF
